dvc/remote/ssh/__init__.py

```diff
@@ -263,19 +262,20 @@ def cache_exists(self, checksums, jobs=None):
         faster than current approach (relying on exists(path_info)) applied in
         remote/base.
         """
-        progress_callback = ProgressCallback(len(checksums))
+        if not self.no_traverse:
+            return list(set(checksums) & set(self.all()))
+
+        with Tqdm(total=len(checksums), unit="md5") as pbar:
 
-        def exists_with_progress(chunks):
-            return self.batch_exists(chunks, callback=progress_callback)
+            def exists_with_progress(chunks):
+                return self.batch_exists(chunks, callback=pbar.update_desc)
 
-        if self.no_traverse:
             with ThreadPoolExecutor(max_workers=jobs or self.JOBS) as executor:
                 path_infos = [self.checksum_to_path_info(x) for x in checksums]
                 chunks = to_chunks(path_infos, num_chunks=self.JOBS)
                 results = executor.map(exists_with_progress, chunks)
                 in_remote = itertools.chain.from_iterable(results)
                 ret = list(itertools.compress(checksums, in_remote))
-                progress_callback.finish("")
                 return ret
 
-        return list(set(checksums) & set(self.all()))
+            pbar.update_desc("", 0)  # clear path name description
```

Comment:
Add `.clear_desc()`? This at least would make the comment not needed.

Reply:
probably need a better description though.. maybe `update_desc("Cache lookup")`?

Reply:
@casperdcl Btw, it actually seems pretty weird here, is it clearing the progress bars from batch_exists? You have another similar line `pbar.update_desc("Checkout", 0)  # clear path name description` up above. Maybe `pbar.finish()` would be better? Is this line used to just leave the trace of the progress bar? If that is the case, we could definitely just drop them, those operations are kinda internal ones, so there is no real need to leave them in the final output, after they've already served the purpose of notifiying user about what is going on.

Reply:
@efiop I wrote this to match the old behavior precisely, where the bar of left behind without and description (the callback would've injected various descriptions). The same is sort of true of the "Checkout" case, where I've replaced a description-less bar + "Checkout finished!" message with simply a completed bar labelled "Checkout".

Basically the new version is identical or slightly less verbose than the original.

Happy to change if you prefer.

Reply:
@casperdcl I understand, and the idea for this PR was precisely that: to make a migration happen and then polish everything in the next PRs, but looks like making a few changes along the way is inevitable. If my understanding of those update_desc() described above is correct, then we could simply remove those lines and be done with this :slightly_smiling_face:

Reply:
Although, we can indeed deal with it on top. Ok.

Reply:
Sure but to be more explicit

old version:
```python
>>> progress_callback.finish("")
[##############################] 100%
```
new version:
```python
>>> pbar.update_desc("", 0)  # clear path name description
100%|██████████| 3/3 [00:01<00:00,  1.96it/s]
```
if removed:
```python
...f17c72d: 100%|██████████| 3/3 [00:01<00:00,  1.96it/s]
```

Actually (unrelated) there's a slight logic error (the line needs to be moved up a little before the function returns).

Reply:
@efiop I made a minor change just before you could merge. It's now rebased on master as e701221cffc7b25a48429d5d449b45849723846d (if you want to cherry-pick or something).

Reply:
@casperdcl Sure, please create a new PR.